### PR TITLE
Improve headings on Roles pages

### PR DESCRIPTION
**Pull request by andysellick**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What
Fix some spacing issues to do with the headings on the roles pages, e.g. https://www.gov.uk/government/ministers/attorney-general

See individual commits for details.

## Why
I stumbled across it and it bothered me.

## Visual changes
Generally improving spacing beneath headings, also replacing the second H2 in 'Current role holder' with a H3, and reducing the font size accordingly.

Before | After
------ | -----
Screenshot 2024-07-12 at 09 27 57 | Screenshot 2024-07-12 at 09 28 02
Screenshot 2024-07-12 at 09 27 47 | Screenshot 2024-07-12 at 09 27 52

### CODE REVIEW

File: app/views/roles/_current_role_holder_with_biography.html.erb [hunk 1]
```diff
@@ -3,10 +3,13 @@
     <%= render "govuk_publishing_components/components/heading", {
       text: t("roles.headings.current_holder"),
       lang: t_fallback("roles.headings.current_holder"),
+      margin_bottom: 2,
     } %>
 
     <%= render "govuk_publishing_components/components/heading", {
       text: role.current_holder["title"],
+      heading_level: 3,
+      font_size: "m",
```
Comment (sairamya93) [inline app/views/roles/_current_role_holder_with_biography.html.erb:12]: what does font-size 'm' mean?

Comment (andysellick) [reply app/views/roles/_current_role_holder_with_biography.html.erb:12]: This is an option for the component, which causes it to display the heading at a medium font size. You can see this documented here: https://components.publishing.service.gov.uk/component-guide/heading/with_margin

Comment (sairamya93) [reply app/views/roles/_current_role_holder_with_biography.html.erb:12]: That is fine

**Review by sairamya93 — APPROVED** [review]
Looks good

**Pull request merged**